EventFilter/GEMRawToDigi/plugins/GEMDigiToRawModule.h

```diff
@@ -0,0 +1,54 @@
+#ifndef EventFilter_GEMDigiToRawModule_h
+#define EventFilter_GEMDigiToRawModule_h
+
+/** \class GEMDigiToRawModule
+ *  \based on CSCDigiToRawModule
+ *  \author J. Lee - UoS
+ */
+
+#include <FWCore/Framework/interface/ConsumesCollector.h>
+#include <FWCore/Framework/interface/EDProducer.h>
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/GEMDigi/interface/GEMDigiCollection.h"
+
+#include "EventFilter/GEMRawToDigi/interface/AMC13Event.h"
+#include "CondFormats/DataRecord/interface/GEMEMapRcd.h"
+#include "CondFormats/GEMObjects/interface/GEMEMap.h"
+#include "CondFormats/GEMObjects/interface/GEMROmap.h"
+
+namespace edm {
+  class ConfigurationDescriptions;
+}
+
+class GEMDigiToRawModule : public edm::EDProducer {
```

Comment:
legacy modules are not allowed

Reply:
this should be of a type known to threaded framework
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes
edm::global would be a better choice if implementation allows, edm::stream is OK.
edm::one is rather an exception and will need some justification.

Reply:
used global, but had to use doBeginRun_ to get up mapping. Is this correct?